In [1]:
import cv2
import numpy as np

In [2]:
def vid2np(path):
    frames = []
    video = cv2.VideoCapture(path)

    cap = cv2.VideoCapture(path)
    ret = True
    while ret:
        ret, img = cap.read() # read one frame from the 'capture' object; img is (H, W, C)
        if ret:
            frames.append(img)
    video = np.stack(frames, axis=0) # dimensions (T, H, W, C)
    return video

In [40]:
gt = vid2np('0_480.mp4')

In [41]:
up = vid2np('0_480_finetunedx2.mp4')

In [42]:
gt.shape

(3600, 480, 856, 3)

In [43]:
up.shape

(3600, 480, 856, 3)

In [44]:
from skimage.metrics import peak_signal_noise_ratio, structural_similarity
import numpy as np

def PSNR(vid1, vid2):
    return np.mean([peak_signal_noise_ratio(image_true, image_test) for image_true, image_test in zip(vid1, vid2)])


def SSIM(vid1, vid2):
    return np.mean([structural_similarity(im1, im2, channel_axis=-1) for im1, im2 in zip(vid1, vid2)])

In [45]:
PSNR(gt, up)

25.248254601145696

In [46]:
SSIM(gt, up)

0.8133833047115111

### Finetuned RealESRGANx2:

PSNR: 25.248

SSIM: 0.813

Скорость обработки 70с

### RealESRGANx4
PSNR: 30.679 \
SSIM: 0.897 \
Скорость обработки 1.30м

### SRCNN (baseline):
PSNR: 24.593 \
SSIM: 0.712 \
Скорость обработки 30с

### SRCNN (baseline)
PSNR: 24.593 \
SSIM: 0.712 \
Скорость обработки 30с